index.js

```diff
@@ -1,21 +1,131 @@
 'use strict';
 const {app, BrowserWindow} = require('electron');
 
-module.exports = async (url, {type = 'json', ...options} = {}) => {
+let mainWindow;
+let isReady = false;
+
+module.exports = async (url, options = {}) => {
 	await app.whenReady();
 
-	const mainWindow = new BrowserWindow({
-		show: false,
-		sandbox: true
+	if (!mainWindow) {
+		mainWindow = new BrowserWindow({
+			show: false,
+			sandbox: true
+		});
+
+		mainWindow.loadURL('about:blank');
+		await mainWindow.webContents.executeJavaScript(ProxyFetch.toString());
+		await mainWindow.webContents.executeJavaScript('window.fetcher = new ProxyFetch();');
```

Comment:
Why two separate calls? `executeJavaScript` is slow. Would be better to have it in one call.